src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,441 @@
+        model_name (str):         Model name to be used in the results file name.
+    Returns:
+        List[str]:                A list of strings that are the file names of the files
+                                  with responses and scores for DK-Bench
+    """
+    validate_output_file_formats(file_formats)
+    response_df = result.dataset.to_pandas()
+    response_df["model_name"] = model_name
+
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    response_df["scores"] = scores
+
+    timestamp = datetime.now().isoformat()
+    response_df["evaluation_run"] = f"run-{timestamp}"
+
+    results_files = []
+    for fmt in file_formats:
+        if IOFileType.JSONL.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.JSONL.value, output_dir, timestamp, model_name
+            )
+            response_df.to_json(f"{results_file}", orient="records", lines=True)
+            results_files.append(results_file)
+            logger.debug(
+                "DK-Bench responses and results written to JSONL file %s", results_file
+            )
+        elif IOFileType.CSV.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.CSV.value, output_dir, timestamp, model_name
+            )
+            response_df.to_csv(f"{results_file}", index=False)
+            results_files.append(results_file)
+            logger.debug(
+                "DK-Bench responses and results written to CSV file %s", results_file
+            )
+        elif IOFileType.XLSX.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.XLSX.value, output_dir, timestamp, model_name
+            )
+            create_excel_results_file(results_file, result)
+            results_files.append(results_file)
+            logger.debug(
+                "DK-Bench responses and results written to Excel file %s", results_file
+            )
+
+    return results_files
+
+
+def is_judge_model_name_valid(judge_model_name: str, api_key: str) -> bool:
+    """
+    Verifies whether or not the judge model provided is the name of a valid
+    OpenAI model to use for the judge in DK-Bench evaluation.
+
+    Valid model names start with "gpt-".
+
+    Args:
+        judge_model_name (str):   Name of the judge model to check validity of.
+        api_key (str):            OpenAI API key.
+    Returns:
+        bool:                     Whether or not the judge_model_name is a valid
+                                  OpenAI model name.
+    """
+    try:
+        client = OpenAI(
+            base_url="https://api.openai.com/v1/",
+            api_key=api_key,
+        )
+        models = client.models.list()
+    except OpenAIError as exc:
+        raise client_utils.ClientException(f"Connection Error {exc}") from exc
+
+    model_ids = [model.id for model in models.data if model.id.startswith("gpt-")]
+    return judge_model_name in model_ids
```

Comment:
You can simplify this entire thing:


```suggestion
    return any(model.id == judge_model_name for model in models.data)

```